File: core/src/main/java/gg/norisk/core/common/NRCPlayer.java
package gg.norisk.core.common;

import java.lang.reflect.Field;
import java.util.Map;
import java.util.Set;
import java.util.UUID;

import gg.norisk.core.channel.ChannelApi;
import gg.norisk.core.manager.HandshakeManager;

public class NRCPlayer {
    
    private static HandshakeManager getHandshakeManager() {
        try {
            Field field = ChannelApi.class.getDeclaredField("handshakeManager");
            field.setAccessible(true);
            return (HandshakeManager) field.get(null);
        } catch (Exception e) {
            throw new RuntimeException("Failed to access handshakeManager", e);
        }
    }
    
    @SuppressWarnings("unchecked")
    public static Set<UUID> listAll() {
        try {
            HandshakeManager manager = getHandshakeManager();
            Field field = manager.getClass().getDeclaredField("nrcPlayers");
            field.setAccessible(true);
            return (Set<UUID>) field.get(manager);
        } catch (Exception e) {
            throw new RuntimeException("Failed to access nrcPlayers", e);
        }
    }
    
    @SuppressWarnings("unchecked")
    public static Set<UUID> listAllNoNRC() {
        try {
            HandshakeManager manager = getHandshakeManager();
            Field field = manager.getClass().getDeclaredField("pendingHandshakes");
            field.setAccessible(true);
            Map<UUID, ?> all = (Map<UUID, ?>) field.get(manager);
            return all.keySet();
        } catch (Exception e) {
            throw new RuntimeException("Failed to access pendingHandshakes", e);
        }
    }
    
    public static boolean user(UUID uuid) {
        return listAll().contains(uuid);
    }
} 

File: core/src/main/kotlin/gg/norisk/core/common/NRCPlayer.kt
package gg.norisk.core.common

import gg.norisk.core.manager.HandshakeManager
import java.util.UUID

object NRCPlayer {
    private val handshakeManager: HandshakeManager
        get() = ChannelApi::class.java.getDeclaredField("handshakeManager").apply { isAccessible = true }.get(null) as HandshakeManager

    fun listAll(): Set<UUID> =
        handshakeManager.javaClass.getDeclaredField("nrcPlayers").apply { isAccessible = true }.get(handshakeManager) as Set<UUID>

    fun listAllNoNRC(): Set<UUID> {
        val all = handshakeManager.javaClass.getDeclaredField("pendingHandshakes").apply { isAccessible = true }.get(handshakeManager) as Map<UUID, *>
        return all.keys
    }

    fun user(uuid: UUID): Boolean = listAll().contains(uuid)
}